transport/ardop/dial.go

```diff
@@ -11,26 +11,61 @@ import (
 	"github.com/la5nta/wl2k-go/transport"
 )
 
-// DialURL dials ardop:// URLs
+// DialURL dials ardop:// URLs.
+//
+// Parameter bw can be used to set the ARQ bandwidth for this connection. See DialBandwidth for details.
 func (tnc *TNC) DialURL(url *transport.URL) (net.Conn, error) {
 	if url.Scheme != "ardop" {
 		return nil, transport.ErrUnsupportedScheme
 	}
-
-	return tnc.Dial(url.Target)
+	bwStr := url.Params.Get("bw")
+	if bwStr == "" {
+		return tnc.Dial(url.Target)
+	}
+	bw, err := BandwidthFromString(bwStr)
+	if err != nil {
+		return nil, err
+	}
+	return tnc.DialBandwidth(url.Target, bw)
 }
 
+// Dial dials a ARQ connection.
 func (tnc *TNC) Dial(targetcall string) (net.Conn, error) {
+	return tnc.DialBandwidth(targetcall, Bandwidth{})
+}
+
+// DialBandwidth dials a ARQ connection after setting the given ARQ bandwidth temporarily.
+//
+// The ARQ bandwidth setting is reverted on any Dial error and when calling conn.Close().
+func (tnc *TNC) DialBandwidth(targetcall string, bw Bandwidth) (net.Conn, error) {
 	if tnc.closed {
 		return nil, ErrTNCClosed
 	}
 
+	var defers []func() error
+	if !bw.IsZero() {
+		currentBw, err := tnc.ARQBandwidth()
+		if err != nil {
+			return nil, err
+		}
+		if err := tnc.SetARQBandwidth(bw); err != nil {
+			return nil, err
+		}
+		defers = append(defers, func() error { return tnc.SetARQBandwidth(currentBw) })
```

Comment:
Interesting, I'm still getting used to treating functions as first-class entities and wouldn't have considered putting them in a slice like this!

Reply:
Yes, I guess it is a bit unfamiliar for some. But it's very handy for injecting specifics in a generic manner, keeping the details nice and isolated while at the same time allowing other parts of the program to execute the functions 🙂 

Somewhat inspired by this: https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis